### PR TITLE
Deprecate and remove this.concatFiles

### LINKED ISSUE #5280: Deprecate and remove this.concatFiles
see https://github.com/ember-cli/ember-cli/issues/5277#issuecomment-167452827

### PR DESCRIPTION
**Pull request by jeffjewiss**
Closes #5280

### CODE REVIEW

**Comment by stefanpenner** [comment]
LGTM

@rwjblue r?

**Comment by rwjblue** [comment]
`this.concatFiles` also adds `options.sourceMapConfig = this.options.sourcemaps;` to the options provided, does `broccoli-concat` do this by default or does that need to be added to each `concat` invocation here?

**Comment by jeffjewiss** [comment]
As far as I can tell `broccoli-concat` does not do this by default. I'll add the sourcemap options to each invocation of `concat`.

**Comment by Turbo87** [comment]
@stefanpenner @rwjblue r?

File: tests/fixtures/dummy-project-outdated/package.json [hunk 1]
```diff
@@ -27,4 +27,4 @@
     "express": "^4.1.1",
     "glob": "^3.2.9"
   }
-}
\ No newline at end of file
+}
```
Comment (stefanpenner) [inline tests/fixtures/dummy-project-outdated/package.json]: ?

Comment (Turbo87) [inline tests/fixtures/dummy-project-outdated/package.json]: most likely an editor that ensures new line at EOF

**Comment by stefanpenner** [comment]
@rwjblue / @Turbo87 do we see anyone using this.concatFiles as a way to monkey patch? Do we care?  If we do, we may need a different deprecation strategy.

**Comment by nathanhammond** [comment]
@stefanpenner Me! I use it to monkey-patch things! I don't disagree with this, but I'm in favor of a different deprecation strategy which still uses `this.concatFiles` for each of the existing invocations.

**Comment by rwjblue** [comment]
We can make a private string that we can pass in options, that is not exported.  That way internal invocations can be "safe", but external ones are not.

**Comment by stefanpenner** [comment]
> We can make a private string that we can pass in options, that is not exported. That way internal invocations can be "safe", but external ones are not.

ya, something like this would do the trick.

**Comment by Turbo87** [comment]
should we close this PR in favor of the new suggested approach?

**Comment by stefanpenner** [comment]
> should we close this PR in favor of the new suggested approach?
>  Hide all checks

should be easy enough to retrofit this PR, @jeffjewiss got any spare cycles for that coming up?

**Comment by jeffjewiss** [comment]
Yeah, I'd be happy to take this on.

I'm having a little trouble understanding the means of deprecating something using a private string that can be passed in to the options. Is there an example of this you could point me to as a reference? Thanks!

**Comment by Turbo87** [comment]
closing in favor of #5926

**Pull request closed without merging**